### ISSUE TITLE
Request - Bookmark management endpoints

### ISSUE THREAD

**Issue by funkydiddykong** [esi/esi-issues#69]
Just the simple ability to rename or delete personal and corporate (If character has permissions) bookmarks by their `id `

**Comment by daniel-lewis-ab** [comment]
We also need to be able to copy them.  For alliance purposes (there are corp bookmarks but not alliance ones?)

**Comment by daniel-lewis-ab** [comment]
We need to be able to copy and paste them from one corp's folder to another's, since we don't expect alliance bookmarks ever.

**Comment by customWOview** [comment]
Agreed.  The ability to export and import bookmarks through the API is a necessary feature to alleviate the alliance bookmark problem.

**Comment by funkydiddykong** [comment]
Corporation bookmarks are currently unable to be viewed using ESI yet (which is weird because anyone in corp can access bookmarks)

I Believe a few Milestones need to be completed for the implementation of this feature:

- [ ] Finish access to corporation bookmarks using ESI
    - Authenticated character must be in the corporation queried 
    - This should return the same information about a bookmark as queried from personal bookmarks

- [ ] A personal deletion endpoint for the deletion of a personal bookmark from its `id`

- [ ] A personal renaming endpoint for the renaming of a personal bookmark from its `id`

- [ ] A corporation deletion endpoint for the deleting corporation bookmarks using `id`
    - Authenticated character must have `Communications Officer` role within the corporation


- [ ] A corporation renaming endpoint for renaming corporation bookmarks using `id`
    - Authenticated character must have `Communications Officer` role within the corporation


Added comments: 
Bookmarks should not be able to be copied by the API because of the following reasons
1. Bookmarks are currently not able to be transferred in-game between corps without trading and as such this would be unfair to players
2. This could be heavy abused by people who create services who use this feature
Bookmarks have value behind them and as such should not be freely replicated or traded out of game

**Comment by customWOview** [comment]
1. Bookmarks are currently not able to be transferred in-game between corps without trading and as such this would be unfair to players without the use of an online service (Anything they can do out of game should be able to be done ingame)
2. This could be heavy abused by people who create services who use this feature
Bookmarks have value behind them and as such should not be freely replicated or traded out of game

I mean are you saying siggy and pathfinder and such tools which benefit those who use them significantly shouldn't be allowed to be used?  These tools allow information sharing on wormhole chains and such out of game.

**Comment by funkydiddykong** [comment]
Siggy and Pathfinder DO NOT share information/assets that cannot be sent with an in-game message.
Examples would be:
1. You can link you current location in-game 
2. You can say what a signature is in the in-game chat
3. You can give someone the information for a wormhole chain 

Basically they just create a user friendly interface for this information that can already be shared ingame.

Siggy CANNOT scan a signature down for you (that in-game information is still up to you) 

By this definition you can already share all your bookmarks by sending the currently available X, Y and Z coordinate data. Or if you want to use the Siggy/Pathfinder approach you can make a website that plots these bookmarks on a map of the solarsystem and that would be "sharing" the same information as Siggy.

**Comment by ccpStuart** [comment]
Duplicate: https://github.com/ccpgames/esi-issues/issues/12

**Comment by daniel-lewis-ab** [comment]
We merged corps over this issue.

**Comment by Slivo-fr** [comment]
+1, delete endpoint for corp bookmarks would be wonderfull

**Comment by lhkbob** [comment]
Would renaming the bookmark fall under valid operations for rearranging the bookmarks?

While it is probably too much to hope for moving a bookmark from a character to *their* own corp's bookmarks, supporting renaming bookmarks could enable some really cool management systems tied to wormhole mapping, etc.

**Comment by Slivo-fr** [comment]
Bump ?

**Comment by Slivo-fr** [comment]
re-bump :P

**Comment by steven-noorbergen** [comment]
I am closing this ticket,

As mentioned in our [dev blog from last year](https://developers.eveonline.com/blog/the-return-of-the-developers-blog), there is currently no plan to restore these endpoints, and we will likely be removing the endpoints from the ESI listing.
